### ISSUE TITLE
Invalid dictionary key in searchindex.js (missing quotes, JS error)

### ISSUE THREAD

**Issue by eudoxos** [sphinx-doc/sphinx#2890]
I am documenting a larger code and searchindex.js contains unquoted UTF-8 token cm³ (63 6D C2 B3), at which Firefox JS engine reports SyntaxError: invalid character, causing any search to stall at "Preparing search...". The  cm³ is used as an unquoted key in the `objtypes` dictionary, but since it is not a valid JS identifier name, it should be quoted when used as key. The full file is available at https://woodem.org/searchindex.js , the stalled search can be reproduced by searching anything at https://woodem.org (at this point of time at least, until sphinx is fixed or I use some workaround to not document that particular object). **UPDATE:** I worked around the issue by not documenting that submodule; therfore the website does not reproduce the issue anymore, but the old searchindex.js is still available from https://woodem.org/searchindex.broken.js)

In terms of the source being documented, the cm³ comes from unicode key (str type in python 3.5) in python dictionary being documented (the keys are passed from c++ using boost::python, I verified their type to be `str`, not `bytes`).

Please quote all keys which are not valid JS identifiers in searchindex.js.

**Comment by shibukawa** [comment]
Could you prepare any small test case? I couldn't reproduce the issue.

**Comment by eudoxos** [comment]
Reproduced using the following MWE (bash script):

```
mkdir /tmp/foobar   
echo "def foobar(arg={'dm³':1e-3}): pass" > /tmp/foobar/foobar.py # note the ³ in the default arg
sphinx-apidoc /tmp/foobar -f -F -o /tmp/foobar-out
perl -pi -e's/:members:/:members: foobar/' /tmp/foobar-out/foobar.rst # explicitly document the foobar function
PYTHONPATH=/tmp/foobar make -C /tmp/foobar-out html
xdg-open /tmp/foobar-out/_build/html/index.html
# try search, will stall
```

**Comment by shibukawa** [comment]
Thanks!

**Comment by tk0miya** [comment]
In my environment, it seems this bug was already fixed with recently release (maybe #2766).
What version do you use? If older than 1.4.5, please upgrade to newest one.

**Comment by eudoxos** [comment]
Confirming #2766 was the culprit, after removing the packaged sphinx version (which was taking precedence for some reason) and updating to 1.4.6, everything works. Sorry for noise & thx for help. Closing.
